src/main/kotlin/it/pagopa/ecommerce/users/services/UserStatisticsService.kt

```diff
@@ -0,0 +1,20 @@
+package it.pagopa.ecommerce.users.services
+
+import it.pagopa.ecommerce.users.documents.LastUsage
+import it.pagopa.ecommerce.users.repositories.UserStatisticsRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+@Service
+class UserStatisticsService(
+    @Autowired private val userStatisticsRepository: UserStatisticsRepository
+) {
+
+    fun findUserLastMethodById(userId: String): Mono<LastUsage> {
+        return userStatisticsRepository
+            .findById(userId)
+            .switchIfEmpty(Mono.error(Exception("User not found")))
```

Comment:
```suggestion
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class UserStatisticsService(
    @Autowired private val userStatisticsRepository: UserStatisticsRepository
) {

    fun findUserLastMethodById(userId: String): Mono<LastUsage> {
        return userStatisticsRepository
            .findById(userId)
            .switchIfEmpty(Mono.error(UserNotFoundException(message = "", cause = null)))
```

use the custom exception here that is mapped to a 404 not found response in exception handler (see [here](https://github.com/pagopa/pagopa-ecommerce-user-stats-service/blob/1c0a86afc1572d73fa9fa539e3390d7cf68ad2a2/src/main/kotlin/it/pagopa/ecommerce/users/exceptions/handlers/ExceptionHandler.kt#L33-L44) and [here](https://github.com/pagopa/pagopa-ecommerce-user-stats-service/blob/main/src/main/kotlin/it/pagopa/ecommerce/users/exceptions/UserNotFoundException.kt))